primitives/evm/src/validation.rs

```diff
@@ -45,17 +45,17 @@ pub struct CheckEvmTransactionConfig<'config> {
 }
 
 #[derive(Debug)]
-pub struct CheckEvmTransaction<'config, E: From<TransactionValidationError>> {
+pub struct CheckEvmTransaction<'config> {
```

Comment:
Look through the whole changes, another approach for your reference:
Add the generic param `T: OnCheckEvmTransaction` on the struct `CheckEvmTransaction`, then you can add another delicate validate function in the struct impl to invoke the `on_check_evm_transaction` there, this approach make the code much clearer when in the validation place.

Reply:
Problem is that `on_check_evm_transaction` requires `&mut CheckEvmTransaction` while other functions require only `&CheckEvmTransaction` .